source/viewmodel.ios.ts

```diff
@@ -272,11 +287,52 @@ class ImagePickerPH extends ImagePicker {
-        PHImageManager.defaultManager().requestImageForAssetTargetSizeContentModeOptionsResultHandler(asset, this._thumbRequestSize, PHImageContentMode.PHImageContentModeAspectFill, this._thumbRequestOptions, function(target, uiImage, info) {
-            var imageSource = new image_source.ImageSource();
-            imageSource.setNativeSource(uiImage);
-            target.setThumb(imageSource);
-        }.bind(this, target));
+        PHImageManager.defaultManager().requestImageForAssetTargetSizeContentModeOptionsResultHandler(asset, this._thumbRequestSize, PHImageContentMode.PHImageContentModeAspectFill,
+            this._thumbRequestOptions, function (target, uiImage, info) {
+                var imageSource = new image_source.ImageSource();
+                imageSource.setNativeSource(uiImage);
+                target.setThumb(imageSource);
+            }.bind(this, target));
+    }
+
+    /**
+     * Creates a new ImageSource from the given image, using the given sizing options.
+     * @param image   The image asset that should be put into an ImageSource.
+     * @param options The options that should be used to create the ImageSource. 
+     */
+    createPHImage(image: PHAsset, options?: ImageOptions): Promise<image_source.ImageSource> {
+        return new Promise<image_source.ImageSource>((resolve, reject) => {
+            var size: CGSize = options ? CGSizeMake(options.maxWidth, options.maxHeight) : PHImageManagerMaximumSize;
+            var resizeMode = PHImageRequestOptions.alloc().init();
+
+            // TODO: Decide whether it is benefical to use PHImageRequestOptionsResizeModeFast
+            //       Accuracy vs Performance. It is probably best to expose these as iOS specific options.
+            resizeMode.resizeMode = PHImageRequestOptionsResizeMode.PHImageRequestOptionsResizeModeExact;
+            resizeMode.synchronous = false;
+
+            // TODO: provide the ability to change this setting.
+            //       Right now, it is needed to make sure that resolve is not called twice.
+            resizeMode.deliveryMode = PHImageRequestOptionsDeliveryMode.PHImageRequestOptionsDeliveryModeHighQualityFormat;
+            resizeMode.normalizedCropRect = CGRectMake(0, 0, 1, 1);
+            PHImageManager.defaultManager().requestImageForAssetTargetSizeContentModeOptionsResultHandler(
+                image,
+                size,
+                PHImageContentMode.PHImageContentModeAspectFill,
+                resizeMode,
+                (createdImage, data) => {
+                    if (createdImage) {
+                        var imageSource = new image_source.ImageSource();
+                        imageSource.setNativeSource(createdImage);
+                        
+                        // TODO: Determine whether runOnRunLoop is needed
+                        //       for callback or not. (See the data() implementation in AssetPH below)
+                        resolve(imageSource);
```

Comment:
@jasssonpet 
The iOS runtime for {N} is associated with the main thread and promises are resolved on the main thread. `runOnRunLoop` is old workaround for the days the {N} promises were not resolved back on the main thread.
It should be safe.